src/v0/destinations/klaviyo/util.js

```diff
@@ -153,22 +99,70 @@ const checkForSubscribe = (message, traitsInfo, destination) => {
 const createCustomerProperties = (message) => {
   let customerProperties = constructPayload(
     message,
-    MAPPING_CONFIG[CONFIG_CATEGORIES.IDENTIFY.name],
+    MAPPING_CONFIG[CONFIG_CATEGORIES.PROFILE.name],
   );
+  customerProperties = removeUndefinedAndNullValues(customerProperties);
+  return customerProperties;
+};
+
+const populateCustomFieldsFromTraits = (message) => {
   // Extract other K-V property from traits about user custom properties
-  customerProperties = extractCustomFields(
+  let customProperties = {};
+  customProperties = extractCustomFields(
     message,
-    customerProperties,
+    customProperties,
     ['traits', 'context.traits'],
     WhiteListedTraits,
   );
-  customerProperties = removeUndefinedAndNullValues(customerProperties);
-  return customerProperties;
+  return customProperties;
+};
+
+const generateBatchedPaylaodForArray = (events) => {
+  let batchEventResponse = defaultBatchRequestConfig();
+  const batchResponseList = [];
+  const metadata = [];
+  // extracting destination from the first event in a batch
+  const { destination } = events[0];
+  // Batch event into dest batch structure
+  events.forEach((ev, index) => {
+    if (index === 0) {
+      batchResponseList.push(ev.message.body.JSON);
+    } else {
+      batchResponseList[0].data.attributes.subscriptions.push(
+        ...ev.message.body.JSON.data.attributes.subscriptions,
+      );
+    }
+    metadata.push(ev.metadata);
+  });
+
+  batchEventResponse.batchedRequest = Object.values(batchEventResponse);
+  batchEventResponse.batchedRequest[0].body.JSON = {
```

Comment:
[Q] We are using `0`th index here on, should we kind of think through scenarios about why only `0`th index is being used ? If we're confident and have a good reason for doing the same, it's fine! Otherwise we need to think through all the scenarios

Reply:
That is done intentionally, as we are creating the `batchedRequest` list above only and it will always have size 1 here.

Reply:
Got it.